src/WebJobs.Script/Description/DotNet/CSharp/Analyzers/InvalidFileMetadataReferenceAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Description.DotNet.CSharp.Analyzers
+{
+    [CLSCompliant(false)]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class InvalidFileMetadataReferenceAnalyzer : DiagnosticAnalyzer
```

Comment:
is the model to add one of these classes for every validation we want to perform? As opposed to us having one for functions, that performs multiple validations